ogr/ogrsf_frmts/wfs/ogroapifdriver.cpp

```diff
@@ -853,6 +874,150 @@ bool OGROAPIFDataset::LoadJSONCollections(const CPLString &osResultIn)
     return !m_apoLayers.empty();
 }
 
+void OGROAPIFDataset::DeterminePageSizeFromAPI(const std::string &itemsUrl)
+{
+    // Try to get max limit from api
+    int nMaximum{-1};
+    int nDefault{-1};
+    // Not sure if min should be considered
+    //int nMinimum { -1 };
+    const CPLJSONDocument &oDoc{GetAPIDoc()};
+    const auto &oRoot = oDoc.GetRoot();
+
+    bool bFound{false};
+
+    // limit from api document
+    if (oRoot.IsValid())
+    {
+
+        const auto paths{oRoot.GetObj("paths")};
+
+        if (paths.IsValid())
+        {
+
+            const auto pathName{itemsUrl.substr(m_osRootURL.length())};
+            const auto path{paths.GetObj(pathName)};
+
+            if (path.IsValid())
+            {
+
+                const auto parameters{path.GetArray("get/parameters")};
+
+                // loop through parameters to find "limit"
+                for (const auto &para : parameters)
+                {
+                    if (para.GetName() == "limit")
+                    {
+                        bFound = true;
```

Comment:
I suspect something is missing here (and in tests), because nMaximum and nDefault are only set in the below !bFound case

Reply:
dead leftover code, the case is handled below (I've removed the `if` as well).